### PR TITLE
fix: improve warning for deprecated equalityFn

### PR DESCRIPTION
**Pull request by dai-shi**
## Related Issues or Discussions

https://github.com/pmndrs/zustand/discussions/1937#discussioncomment-6639426
https://github.com/pmndrs/zustand/discussions/1937#discussioncomment-6643081
https://github.com/pmndrs/zustand/discussions/1937#discussioncomment-6646040

Follow-up #1945

## Summary

Improve the warning message based on the feedback.

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: src/react.ts [hunk 1]
```diff
@@ -47,10 +49,15 @@ export function useStore<TState, StateSlice>(
...
+    import.meta.env?.MODE !== 'production' &&
+    equalityFn &&
+    !didWarnAboutEqualityFn
+  ) {
     console.warn(
-      "[DEPRECATED] Use `createWithEqualityFn` from 'zustand/traditional'. https://github.com/pmndrs/zustand/discussions/1937"
+      "[DEPRECATED] Use `createWithEqualityFn` instead of `create` or use `useStoreWithEqualityFn` instead of `useStore`. They can be imported from 'zustand/traditional'. https://github.com/pmndrs/zustand/discussions/1937"
     )
+    didWarnAboutEqualityFn = true
   }
```
Comment (sewera) [inline src/react.ts:61]: It may be a good idea to extract this to `equalityFn && warnAboutEqualityFnOnce()`

Comment (dai-shi) [reply src/react.ts:61]: Unfortunately, that doesn't work with minification/dead-code-elimination in production.

Comment (sewera) [reply src/react.ts:61]: I see, I don't have any more suggestions then.

**Review by sewera — APPROVED** [review]

**Comment by dai-shi** [comment]
I'm still waiting for some 👍 feedback in #1937.
Will merge it anyway tomorrow or so.

**Pull request merged**